lib/ae_mdw_web/plugs/paginated_plug.ex

```diff
@@ -83,4 +128,75 @@ defmodule AeMdwWeb.Plugs.PaginatedPlug do
         end
     end
   end
+
+  defp extract_query(query_params) do
+    query_params
+    |> Enum.reject(fn {key, _val} -> key in @pagination_param_keys end)
+    |> Enum.reduce_while({:ok, %{}}, fn {key, val}, {:ok, top_level} ->
+      kw = (key in @type_query_params && :types) || :ids
+      group = Map.get(top_level, kw, MapSet.new())
+
+      case extract_group(key, val, group) do
+        {:ok, new_group} -> {:cont, {:ok, Map.put(top_level, kw, new_group)}}
+        {:error, reason} -> {:halt, {:error, reason}}
+      end
+    end)
+  end
+
+  defp extract_group("type", val, group) do
+    case Validate.tx_type(val) do
+      {:ok, type} -> {:ok, MapSet.put(group, type)}
+      {:error, {err_kind, offender}} -> {:error, AeMdw.Error.to_string(err_kind, offender)}
+    end
+  end
+
+  defp extract_group("type_group", val, group) do
+    case Validate.tx_group(val) do
+      {:ok, new_group} ->
+        {:ok, new_group |> Node.tx_group() |> MapSet.new() |> MapSet.union(group)}
+
+      {:error, {err_kind, offender}} ->
+        {:error, AeMdw.Error.to_string(err_kind, offender)}
+    end
+  end
+
+  defp extract_group(key, val, group) do
+    {_is_base_id?, validator} = AeMdw.Db.Stream.Query.Parser.classify_ident(key)
+
+    try do
+      {:ok, MapSet.put(group, {key, validator.(val)})}
+    rescue
+      err in [AeMdw.Error.Input] ->
+        {:error, err.message}
+    end
+  end
+
+  defp default_forward_range, do: first_gen()..last_gen()
+
+  defp default_backward_range, do: last_gen()..first_gen()
+
+  defp first_gen do
+    case Mnesia.first_key(Block, nil) do
+      nil -> 0
+      {kbi, _txi} -> kbi
+    end
+  end
+
+  defp last_gen do
+    case Mnesia.last_key(Block, nil) do
+      nil -> 0
+      {kbi, _txi} -> kbi
+    end
+  end
```

Comment:
Can move to `AeMdw.Db.Util` and rename existing ones to `first_gen!` and `last_gen!`

Reply:
I've moved these to the Util module, and re-adjusted credo and all of the tests that were mocking it. Not sure why integration tests are failing for you, though, they never failed for me.